FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Authentication/ITokenCredentialProvider.cs

```diff
@@ -0,0 +1,14 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Azure.Core;
```

Comment:
It seems that `Common` project is about the definition, is it a good solution to add authentication here?